Sources/FluentKit/Model/Model+CRUD.swift

```diff
@@ -6,96 +6,101 @@ extension Model {
             return self.create(on: database)
         }
     }
-
+    
     public func create(on database: Database) -> EventLoopFuture<Void> {
+        let responder = BasicModelResponder { (_, _, db) in self._create(on: db) }
+        return database.context.middleware.makeResponder(chainingTo: responder)
```

Comment:
Maybe we could wrap the `BasicModelResponder` in an extension to get syntax like this:

```swift
public func create(on database: Database) -> EventLoopFuture<Void> {
    database.context.middleware.chainingTo { event, model, db in 
        self._create(on: db)
    }.handle(.create, self, on: database)
}
```

Reply:
This brings up an interesting idea, what if we had a method like `handle` on model that accepted which model event to perform? Then we could do something like:

```swift
public func create(on database: Database) -> EventLoopFuture<Void> {
    database.context.middleware.chainingTo { event, model, db in 
        model.handle(event, on: db)
    }.handle(.create, self, on: database)
}
```
This would allow for the middleware to fully change the `model` and `event`.